inception-concept-linking/src/main/java/de/tudarmstadt/ukp/inception/conceptlinking/util/QueryUtil.java

```diff
@@ -164,15 +164,23 @@ private static String getFullTextMatchingQueryPartDefault(int aLimit)
     private static String getFullTextMatchingQueryPartLucene(int aLimit)
     {
         return  String.join("\n",
-            "    SELECT DISTINCT ?e2 ?altLabel ?description WHERE",
+            "    SELECT DISTINCT ?iri ?altLabel ?description WHERE",
             "    {",
-            "      ?e2 search:matches ?match .",
-            "      ?match search:query ?string ;",
-            "             search:property ?labelIri ;",
-            "             search:snippet ?altLabel",
+            "      {  ?iri ?a ?concept .",
+            "          FILTER NOT EXISTS { ?iri ?labelIri [] }",  // only concepts without label
+            "          FILTER regex( str(?iri) , ?string, \"i\") ",  // case-insensitive matching
```

Comment:
Here, we are no longer using full-text search. Using `regex` does not really scale. Have you tried this with large-scale KBs like Wikidata on Virtuoso?

Reply:
This query snippet is only used for local knowledge bases that are indexed by Lucene.

Reply:
Ok, but Lucene dos not speed up `regex` - so even for local KBs, this will be quite slow unless the KB is really small. Using `regex` is *not* a full text query as the title of the method suggests. The old code was using the full-text index from Lucene, but this new code does not.